### PR TITLE
Positive values only for item.unitPrice and item.units

### LINKED ISSUE #32: Item.unitPrice doesn't allow negative values?
**Describe the bug**

- First, document [https://docs.paytrail.com/#/?id=item](https://docs.paytrail.com/#/?id=item) doesn't say that `item.unitPrice` must have minimum value of 0.

- Second, old [`E2 Interface`](https://old-docs.paytrail.com/payments/e2-interface/fields/) field `ITEM_UNIT_PRICE` allowed prices to be negative. You ask why this is needed? Well in our store (Zen Cart) user can use his/hers accounts `tokens` in their order. These tokens are acquired by gift vouchers and are processed as negative items in the order. Other items that fall in to this category are `customer group discounts` and `coupons` that make discounts from the totals etc. They both are also processed as negative items.

Here's payload that fails:
```
[stamp] => 69-691805221041180
[reference] => 691805221041180
[amount] => 1750
[currency] => EUR
[language] => FI
[orderId] => 69
[items] => Array
(
    [0] => Array
        (
            [unitPrice] => 850
            [units] => 1
            [vatPercentage] => 24
            [productCode] => A123456789
            [description] => Asetoni 1 l
        )

    [1] => Array
        (
            [unitPrice] => 1000
            [units] => 1
            [vatPercentage] => 24
            [productCode] => -1
            [description] => Posti (Kotipaketti - kotiovelle klo 21 mennessä)
        )

    [2] => Array
        (
            [unitPrice] => -100
            [units] => 1
            [vatPercentage] => 0
            [productCode] => -1
            [description] => Tilin varat
        )

)
..
```

Paytrail Response:
```
Array (
    [status] => error
    [message] => Request payload schema validation failed. Validate request against the provided schema.
    [meta] => Array
        (
            [0] => instance.items[2].unitPrice must have a minimum value of 0
        )

)
```

**Question: How can I get this to work with new API?**
If I take this negative `items[2]` out of the payload or zero it, then the `amount` is not going to match total sum of items.

```
  [status] => error
  [message] => Sum of purchase item amounts does not match total amount.
```

**Expected behaviour**

`item.unitPrice` should also allow negative values.

**Actual behaviour**

`item.unitPrice` must have a minimum value of 0.

### PR DESCRIPTION
**Pull request by loueranta-paytrail**
## Related tickets & documents

Resolves #32

## Description

Added note that API supports only positive values for `item.unitPrice` and `item.units`.

### CODE REVIEW

File: docs/README.md [hunk 1]
```diff
@@ -917,8 +917,8 @@ General API HTTP status codes and what to expect of them.
...
-| unitPrice     | integer                   | <center>x</center> | 1000                                 | Price per unit, in each country's minor unit, e.g. for Euros use cents. By default price should include VAT, unless [`usePricesWithoutVat`](#request-body) is set to true. |
-| units         | integer                   | <center>x</center> | 5                                    | Quantity, how many items ordered                                                                                                                                           |
+| unitPrice     | integer                   | <center>x</center> | 1000                                 | Price per unit, in each country's minor unit, e.g. for Euros use cents. By default price should include VAT, unless [`usePricesWithoutVat`](#request-body) is set to true. Only positive values accepted. |
+| units         | integer                   | <center>x</center> | 5                                    | Quantity, how many items ordered. Only positive values supported.                                                                                                                                          |
```
Comment (mikkorantalainen) [inline docs/README.md:921]: Is zero okay here?

File: docs/README.md [hunk 2]
```diff
@@ -917,8 +917,8 @@ General API HTTP status codes and what to expect of them.
...
 | ------------- | ------------------------- | ------------------ | ------------------------------------ | -------------------------------------------------------------------------------------------------------------------------------------------------------------------------- |
-| unitPrice     | integer                   | <center>x</center> | 1000                                 | Price per unit, in each country's minor unit, e.g. for Euros use cents. By default price should include VAT, unless [`usePricesWithoutVat`](#request-body) is set to true. |
-| units         | integer                   | <center>x</center> | 5                                    | Quantity, how many items ordered                                                                                                                                           |
+| unitPrice     | integer                   | <center>x</center> | 1000                                 | Price per unit, in each country's minor unit, e.g. for Euros use cents. By default price should include VAT, unless [`usePricesWithoutVat`](#request-body) is set to true. Only positive values accepted. |
```
Comment (mikkorantalainen) [inline docs/README.md:920]: Should zero be accepted here to allow including zero cost campaign items in the receipt?

File: docs/README.md (hunk 1 above)
Comment (loueranta-paytrail) [reply docs/README.md]: Actually, in some cases zero can be used: if the total amount of all items is more than zero, then some of the items can be set to zero.

File: docs/README.md (hunk 2 above)
Comment (loueranta-paytrail) [reply docs/README.md]: Yes, if the total amount of all items is more than zero, then some of the items can be set to zero.

**Review by jfrojd-paytrail — APPROVED** [review]

**Pull request merged**